tests/tests/src/lampo_tests.rs

```diff
@@ -184,205 +184,97 @@ pub async fn pay_offer_simple_case_lampo() -> error::Result<()> {
-                amount: 1_000_000,
-                public: true,
-                addr: Some("127.0.0.1".to_owned()),
-                port: Some(node2.port),
+            "invoice",
+            request::GenerateInvoice {
+                description: "test decode".to_owned(),
+                amount_msat: Some(100_000),
+                expiring_in: None,
             },
         )
-        .unwrap();
-    assert!(response.get("tx").is_some());
-
-    wait!(|| {
-        while let Ok(event) = events.recv_timeout(Duration::from_millis(10)) {
-            node2.fund_wallet(6).unwrap();
-            if let Event::Lightning(LightningEvent::ChannelReady {
-                counterparty_node_id,
-                ..
-            }) = event
-            {
-                if counterparty_node_id.to_string() == node1.info.node_id {
-                    return Err(());
-                }
-                return Ok(());
-            };
-            // check if lampo see the channel
-            let channels: response::Channels =
-                node2.lampod().call("channels", json::json!({})).unwrap();
-            if channels.channels.is_empty() {
-                return Err(());
-            }
-
-            if !channels.channels.first().unwrap().ready {
-                return Err(());
-            }
-
-            let channels: response::Channels =
-                node1.lampod().call("channels", json::json!({})).unwrap();
-
-            if channels.channels.is_empty() {
-                return Err(());
-            }
-
-            if channels.channels.first().unwrap().ready {
-                return Ok(());
-            }
-        }
-        node2.fund_wallet(6).unwrap();
-        Err(())
-    });
-
-    let offer: response::Offer = node2.lampod().call(
-        "offer",
-        request::GenerateOffer {
-            description: None,
-            amount_msat: None,
-        },
-    )?;
+        .await?;
 
-    log::info!(target: &node2.info.node_id, "offer generated `{:?}`", offer);
+    log::info!(target: &node2.info.node_id, "invoice generated `{:?}`", invoice);
 
-    let decode: response::Bolt12InvoiceInfo = node2.lampod().call(
-        "decode",
-        request::DecodeInvoice {
-            invoice_str: offer.bolt12,
-        },
-    )?;
+    let decode: response::Bolt11InvoiceInfo = node2
+        .lampod()
+        .call(
+            "decode",
+            request::DecodeInvoice {
+                invoice_str: invoice.bolt11.clone(),
+            },
+        )
+        .await?;
 
     assert_eq!(decode.issuer_id.clone(), Some(node2.info.node_id.clone()));
     log::info!(target: &node2.info.node_id, "decode offer `{:?}`", decode);
+
+    let pay: response::PayResult = node1
+        .lampod()
+        .call(
+            "pay",
+            request::Pay {
+                invoice_str: invoice.bolt11,
+                amount: None,
+                bolt12: None,
+            },
+        )
+        .await?;
```

Comment:
can you assert on some field of the pay result? 

like `assert_eq!(pay.state, PaymentState::Success)`